File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/store/StoreImageCreator.java
package com.depromeet.threedollar.domain.rds.user.domain.store;

import org.jetbrains.annotations.NotNull;

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture;

import lombok.AccessLevel;
import lombok.Builder;
import lombok.NoArgsConstructor;

@NoArgsConstructor(access = AccessLevel.PRIVATE)
@TestFixture
public class StoreImageCreator {

    @Builder
    public static StoreImage create(
        @NotNull Store store,
        @NotNull Long userId,
        @NotNull String url
    ) {
        return StoreImage.builder()
            .store(store)
            .userId(userId)
            .url(url)
            .build();
    }

}

File: threedollar-domain/threedollar-domain-rds/src/testFixtures/java/com/depromeet/threedollar/domain/rds/user/domain/store/StoreImageCreator.kt
package com.depromeet.threedollar.domain.rds.user.domain.store

import com.depromeet.threedollar.domain.rds.user.domain.TestFixture

@TestFixture
object StoreImageCreator {

    @JvmOverloads
    @JvmStatic
    fun create(
        storeId: Long,
        userId: Long,
        url: String,
        status: StoreImageStatus = StoreImageStatus.ACTIVE
    ): StoreImage {
        return StoreImage.builder()
            .storeId(storeId)
            .userId(userId)
            .url(url)
            .status(status)
            .build()
    }

}
